src/transformers/training_args.py

```diff
@@ -211,7 +211,17 @@ def to_sanitized_dict(self) -> Dict[str, Any]:
         Sanitized serialization to use with TensorBoard’s hparams
         """
         d = dataclasses.asdict(self)
+        d = {
+            **d,
+            **{
+                "train_batch_size": self.train_batch_size,
+                "eval_batch_size": self.eval_batch_size,
+                "n_gpu": self.n_gpu,
```

Comment:
I think some attention needs to be paid here. When using distributed training, `self.n_gpu` would always be `1`, which causes ambiguity. Is it expected behavior?

Reply:
Hm, that's a good point. Maybe it's best to just leave out `n_gpu`. 

The purpose of this feature was to have all the necessary hparams to reproduce a result. I've been training models between a system with one gpu, and another one with multiple gpus, which meant unless I knew which system a model was trained on, I wouldn't know what the true batch size it was trained with.

We don't really need `n_gpu` to reproduce the result so I'll just remove it from the dict.